std/range/primitives.d

```diff
@@ -176,6 +170,46 @@ template isInputRange(R)
     static assert( isInputRange!(inout(int)[]));
 }
 
+/**
+Helper function for $(D isInputRange). Can also be used to statically
+assert that a user-defined type is an input range.
```

Comment:
Right. The main problem here is that the unit tests fail because the code is actually executed (although it shouldn't) for interfaces. I need to find a better way of dealing with this.